lib/ansible/modules/cloud/vmware/vmware_guest_network.py

```diff
@@ -0,0 +1,453 @@
+                if 'name' in network and not self.get_network_by_name(self.content, network['name']):
+                    self.module.fail_json(msg="Network '%(name)s' does not exist." % network)
+                elif 'vlan' in network:
+                    objects = get_all_objs(self.content, [vim.dvs.DistributedVirtualPortgroup])
+                    dvps = [x for x in objects if to_text(self.get_parent_datacenter(x).name) == to_text(self.params['datacenter'])]
+                    for dvp in dvps:
+                        if hasattr(dvp.config.defaultPortConfig, 'vlan') and \
+                                isinstance(dvp.config.defaultPortConfig.vlan.vlanId, int) and \
+                                str(dvp.config.defaultPortConfig.vlan.vlanId) == str(network['vlan']):
+                            network['name'] = dvp.config.name
+                            break
+                        if 'dvswitch_name' in network and \
+                                dvp.config.distributedVirtualSwitch.name == network['dvswitch_name'] and \
+                                dvp.config.name == network['vlan']:
+                            network['name'] = dvp.config.name
+                            break
+                        if dvp.config.name == network['vlan']:
+                            network['name'] = dvp.config.name
+                            break
+                    else:
+                        self.module.fail_json(msg="VLAN '%(vlan)s' does not exist." % network)
+
+                if 'device_type' in network and network['device_type'] not in self.nic_device_type.keys():
+                    self.module.fail_json(msg="Device type specified '%s' is invalid. "
+                                              "Valid types %s " % (network['device_type'], self.nic_device_type.keys()))
+
+                if 'mac' in network and not self.is_valid_mac_addr(network['mac']):
+                    self.module.fail_json(msg="Device MAC address '%s' is invalid. "
+                                              "Please provide correct MAC address." % network['mac'])
+
+                network_list.append(network)
+
+        return network_list
+
+    def get_network_config_spec(self, vm_obj, network_list):
+        # create network adapter config spec for adding, editing, removing
+        for network in network_list:
+            # add new network adapter
+            if network['state'].lower() == 'new':
+                nic_spec = self.create_network_adapter(network)
+                nic_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
+                self.change_detected = True
+            # reconfigure network adapter or remove network adapter
+            else:
+                nic_device = None
+                if 'label' in network:
+                    nic_device = self.get_network_device_by_label(vm_obj, device_label=network['label'])
+                elif 'label' not in network and 'device_type' in network:
```

Comment:
Why not with `mac`?

Reply:
set "mac" as another parameter to find the network adapter to do reconfigure or remove, besides "label", "device_type". Thank you.